core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -387,7 +391,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         @SuppressWarnings("unchecked")
         public static Suggestion<? extends Entry<? extends Option>> fromXContent(XContentParser parser) throws IOException {
             ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
-            return XContentParserUtils.parseTypedKeysObject(parser, Aggregation.TYPED_KEYS_DELIMITER, Suggestion.class);
+            SetOnce<Suggestion> suggestion = new SetOnce<>();
```

Comment:
nit: would it work to use Optional here instead? then fromXContent could return an Optional in this case and we would check isPresent instead of the null check above?